k8s/pkg/apis/longhorn/v1beta2/objectstore.go

```diff
@@ -163,6 +160,10 @@ type ObjectStoreEndpointSpec struct {
 
 	// +optional
 	DomainName string `json:"domainName,omitempty"`
+
+	// Reference to a secret containing TLS certificate and key for DomainName
+	// +optional
+	TLS corev1.SecretReference `json:"tls,omitempty"`
```

Comment:
This function is new compared to the design in our LEP. At the beginning, we agreed to let user handle ingress/TLS configuration by themself if they want to expose the s3 ObjectStore out of the cluster. Now, we are doing this inside Longhorn manager controller, this could complicated the implementation as:
1. What should Longhorn choose for the ingressclass?
2. How do we handle the backup/restore of the TLS secret?
3. How do we handle the certificate expeation/renewal?

Reply:
Marking this as resolved. we dicussed in the Slack channel that we will proxy the traffic to S3gw UI via Longhorn UI so there is no need for creating ingress for UI. The overall design is:

> 1. By default, we don't deploy any ingress (either for longhorn-ui, s3 ObjectStore Ui, s3 ObjectStore backend). Users are not required to have ingress controller deployed inside the cluster.
> 1. If user specifically ask for ingress to s3 ObjectStore backend (by specifying it in the spec.Endpoints, Longhorn controller can create the ingresses to S3 ObjectStore backend. In this case, user need to make sure that they have ingress controller inside the cluster.
> 1. User can access Longhorn UI and s3 ObjectEndpoint without any ingress. (e.g., k port-forward -n longhorn-system svc/longhorn-frontend 8080:80  ). Longhorn UI pod will act as reverse proxy to forward the traffic to s3gw UI internal service.